test/unit/conditions/compound-condition.test.ts

```diff
@@ -1,85 +1,121 @@
-import { CompoundCondition } from '../../../src/conditions';
-import { ERC721Ownership } from '../../../src/conditions/predefined/erc721';
+import { Condition } from '../../../src/conditions';
+import { CompoundCondition } from '../../../src/conditions/base';
+import { compoundConditionSchema } from '../../../src/conditions/compound-condition';
 import {
   testContractConditionObj,
   testRpcConditionObj,
   testTimeConditionObj,
 } from '../testVariables';
 
-describe('validate', () => {
-  const ownsBufficornNFT = ERC721Ownership.fromObj({
-    contractAddress: '0x1e988ba4692e52Bc50b375bcC8585b95c48AaD77',
-    parameters: [3591],
-    chain: 5,
-  }).toObj();
-
+describe('validation', () => {
   it('accepts or operator', () => {
-    const orCondition = new CompoundCondition({
+    const conditionObj = {
       operator: 'or',
-      operands: [ownsBufficornNFT, testTimeConditionObj],
-    }).toObj();
+      operands: [testContractConditionObj, testTimeConditionObj],
+    };
+    const result = Condition.validate(compoundConditionSchema, conditionObj);
 
-    expect(orCondition.operator).toEqual('or');
-    expect(orCondition.operands).toEqual([
-      ownsBufficornNFT,
-      testTimeConditionObj,
-    ]);
+    expect(result.error).toBeUndefined();
+    expect(result.data).toEqual({
+      ...conditionObj,
+      conditionType: 'compound',
```

Comment:
For the documentation and Conditions Schema, should we consider `compound` to be a fourth condition type, or on a different 'level' from `contract`, `rpc` and `time`? And does the constructor tell the developer that they're creating a compound condition set?

Reply:
I think `CompoundCondition` is somewhat different since (by definition) it can't be used on its own. That's why I didn't put it into `conditions/base` which is supposed to contain some basic building blocks for conditions.

Yes, the constructor expects `CompoundConditionProps` and provides type hints for TS developers.